darts/metrics/metrics.py

```diff
@@ -11,14 +11,36 @@
 from ..utils.statistics import check_seasonality
 from ..logging import raise_if_not, get_logger
 from warnings import warn
-from typing import Optional
+from typing import Optional, Callable
+from inspect import signature
 
 logger = get_logger(__name__)
 
 
+def multivariate_support(func):
+    """
+    This decorator transforms a metric function that takes as input two univariate TimeSeries instances
+    into a function that takes two equally-sized multivariate TimeSeries instances, computes the pairwise univariate
+    metrics for components with the same indices, and returns a float value that is computed as a function of all the
+    univariate metrics using a `reduction` subroutine passed as argument to the metric function.
+    """
+
+    def wrapper_multivariate_support(*args, **kwargs):
+        raise_if_not(args[0].width == args[1].width, "The two TimeSeries instances must have the same width.", logger)
+        value_list = []
+        for i in range(args[0].width):
+            value_list.append(func(args[0].univariate_component(i), args[1].univariate_component(i),
```

Comment:
will this work if some arguments are given by keyword (in kwargs) instead of being given in args?

Reply:
Right now the two TimeSeries instances have to be given in args, not in kwargs. Is this a reasonable assumption?

Reply:
I think it would be a little safer to check whether they are in the kwargs first (and if yes, use the kwargs ones).

Reply:
done